api/pkg/service/auth/auth.go

```diff
@@ -139,6 +140,10 @@ func (r *request) addUser(ghUser *github.User) (*model.User, error) {
 		user.GithubName = ghUser.GetName()
 		user.Type = model.NormalUserType
 
+		if user.AvatarURL == "" {
```

Comment:
this should be outside of `if user.GithubName == "" {}`

Reply:
```
        // User already exist, check if GitHub Name is empty
	// If Name is empty, then user is inserted through config.yaml
	// Update user with remaining details

	if user.GithubName == "" {
		user.GithubName = ghUser.GetName()
		user.Type = model.NormalUserType
                user.AvatarURL = ghUser.GetAvatarURL()

		if err = r.db.Save(&user).Error; err != nil {
			r.log.Error(err)
			return nil, err
		}
	}
	// For existing user, check if URL is not added
        if user.AvatarURL == "" {
		user.AvatarURL = ghUser.GetAvatarURL()
		if err = r.db.Save(&user).Error; err != nil {
		      r.log.Error(err)
		      return nil, err
		  }
	}


```

Reply:
also, could you please add a test for this.. where user in DB doesn't have `URL` but after login URL is added